### PR TITLE
Add grpc example

### PR DESCRIPTION
**Pull request by kaibocai**
Add Grpc example using [opentelemetry-grpc-1.6](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/grpc-1.6/library)

### CODE REVIEW

**Comment by kaibocai** [comment]
cc @trask

File: grpc/src/main/proto/helloworld.proto [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.kc";
```
Comment (trask) [inline grpc/src/main/proto/helloworld.proto:8]:
```suggestion
option java_package = "io.opentelemetry.examples.grpc";
```

File: grpc/src/main/proto/helloworld.proto [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+option java_outer_classname = "HelloWorldProto";
+option objc_class_prefix = "HLW";
```
Comment (trask) [inline grpc/src/main/proto/helloworld.proto:10]:
```suggestion
```

**Review by trask — COMMENTED** [review]
can you add a README.md?

File: grpc/src/main/proto/helloworld.proto (hunk 1 above)
Comment (kaibocai) [reply grpc/src/main/proto/helloworld.proto]: updated!

File: grpc/src/main/proto/helloworld.proto (hunk 2 above)
Comment (kaibocai) [reply grpc/src/main/proto/helloworld.proto]: removed!

**Comment by kaibocai** [comment]
Added README file!

**Comment by kaibocai** [comment]
Hi @trask , I got a question want to check using [opentelemetry-grpc-1.6](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/grpc-1.6/library), it automatically inject the trace context to the grpc call and trace the grpc request from both client and server side. I was wondering does this library also provide a way to get the client trace context so that I can build child span from it on server side. Thanks!

**Comment by trask** [comment]
> Hi @trask , I got a question want to check using [opentelemetry-grpc-1.6](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/grpc-1.6/library), it automatically inject the trace context to the grpc call and trace the grpc request from both client and server side. I was wondering does this library also provide a way to get the client trace context so that I can build child span from it on server side. Thanks!

the library instrumentation on the server side should be setting the parent (client) span context as active, so that any spans you create on the server side should automatically parent the client span

File: grpc/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# GRPC Example
+
+**Note:** This is an advanced scenario useful for people that want to *manually* instrument their own code.
```
Comment (breedx-splk) [inline grpc/README.md:3]:
```suggestion
**Note:** This is an advanced scenario useful for people who want to *manually* instrument their own code.
```

File: grpc/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+This is a simple example that demonstrates how to use the OpenTelemetry SDK
+to *manually* instrument a simple GRPC based Client/Server application.
+The example utilize the [opentelemetry-grpc-1.6](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/grpc-1.6/library#library-instrumentation-for-grpc-160)
```
Comment (breedx-splk) [inline grpc/README.md:7]:
```suggestion
Ths example utilizes the [opentelemetry-grpc-1.6](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/grpc-1.6/library#library-instrumentation-for-grpc-160)
```

File: grpc/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+to instrument both the GRPC client and server. 
+
+# How to run
+
+## Prerequisites
+* Java 1.8.231
```
Comment (breedx-splk) [inline grpc/README.md:13]: That's oddly specific. Can it run with later versions like 11 or 17 as well?

File: grpc/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+* Be on the project root folder
```
Comment (breedx-splk) [inline grpc/README.md:14]: People might think that they need to be in the repo root, not this `grpc` subdir. Can we clarify?

File: grpc/src/main/java/io/opentelemetry/examples/grpc/HelloWorldClient.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.logging.Logger;
+
+/** A simple client that requests a greeting from the {@link HelloWorldServer}. */
+public class HelloWorldClient {
+
+  private static final Logger logger = Logger.getLogger(HelloWorldClient.class.getName());
+  private final ManagedChannel channel;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+  private static final OpenTelemetry openTelemetry =
+      ExampleConfiguration.initializeOpenTelemetry("localhost", 9411);
```
Comment (breedx-splk) [inline grpc/src/main/java/io/opentelemetry/examples/grpc/HelloWorldClient.java:25]: nit: I like to have the static fields grouped above the instance fields.

**Review by breedx-splk — APPROVED** [review]
I think this is a great start, thanks for the contribution!

I appreciate that you call this an advanced topic for *manual* instrumentation, but I think another sentence in the README that explains that if you're using the agent that you get this instrumentation "out-of-the-box" and that this isn't necessary. I just don't want a casual user to think that they _need_ to do this when the agent is available.

File: grpc/README.md (hunk 5 above)
Comment (kaibocai) [reply grpc/README.md:13]: I copied this from other examples, seems it's out of date and I am running on java17. Have it removed!

File: grpc/build.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    }
+}
+
+var grpcVersion = "1.60.1" // CURRENT_GRPC_VERSION
+var protobufVersion = "3.25.1"
+var protocVersion = protobufVersion
+
+dependencies {
+    runtimeOnly ("io.grpc:grpc-netty-shaded:1.59.1")
+    implementation ("io.grpc:grpc-protobuf:1.59.1")
```
Comment (trask) [inline grpc/build.gradle.kts:24]:
```suggestion
    implementation ("io.grpc:grpc-protobuf:${grpcVersion}")
```

File: grpc/build.gradle.kts [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    implementation("io.opentelemetry:opentelemetry-api")
+    implementation("io.opentelemetry:opentelemetry-sdk")
+    implementation("io.opentelemetry:opentelemetry-exporter-zipkin")
+    implementation("io.opentelemetry.instrumentation:opentelemetry-grpc-1.6")
+
+    //alpha modules
+    implementation("io.opentelemetry.semconv:opentelemetry-semconv")
+
+    implementation("com.google.j2objc:j2objc-annotations:2.8")
+    implementation("org.apache.tomcat:annotations-api:6.0.53")
```
Comment (trask) [inline grpc/build.gradle.kts:40]: are these used?
```suggestion
```

**Review by trask — APPROVED** [review]
thx!

**Pull request merged**